nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -585,6 +618,32 @@ in {
       path = with pkgs; [ file imagemagick ffmpeg ];
     };
 
+    systemd.services.mastodon-media-auto-remove = lib.mkIf cfg.mediaAutoRemove.enable {
+      description = "Mastodon media auto remove";
+      environment = env;
+      serviceConfig = {
+        Type = "oneshot";
+        # Remove remote media attachments and preview cards older than one month.
+        ExecStart = let
```

Comment:
I also think you should use `script =` here instead of `ExecStart`. It makes the `writeShellScriptBin` call redundant (and even has the neat property of being composable).